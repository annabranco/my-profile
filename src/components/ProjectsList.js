import React from 'react';
import projectsDB from '../db/projectsDB';

class ProjectsList extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      showThumbnails: false
    };
  }

  handleThumbnails = e => {
    this.setState({
      showThumbnails: e.currentTarget.checked
    });

    if (!window.matchMedia('(min-width: 768px)').matches) {
      if (!this.props.seeAll) {
        if (e.currentTarget.checked) {
          document.querySelector('.projects__list').style.height = '400px';
        } else {
          document.querySelector('.projects__list').style.height = '210px';
        }
      }
    }

    this.props.handleAdjustThumbnailsView(e.currentTarget.checked);
  };

  nextProjects = () => {
    let adjustments = 1;
    if (this.state.showThumbnails) {
      adjustments = 2;
    }

    if (window.matchMedia('(min-width: 768px)').matches) {
      document.querySelector('.projects__list').scrollTop += 248;
    } else {
      if (this.state.showThumbnails) {
        document.querySelector('.projects__list').scrollTop +=
          195 * adjustments;
      } else {
        document.querySelector('.projects__list').scrollTop +=
          248 * adjustments;
      }
    }

    document
      .querySelector('.fa-arrow-alt-circle-up')
      .classList.remove('invisible');
    document
      .querySelector('.project__seeMore-up')
      .classList.remove('invisible');

    if (window.matchMedia('(min-width: 768px)').matches) {
      if (
        document.querySelector('.projects__list').scrollTop >=
        744 * adjustments
      ) {
        document
          .querySelector('.fa-arrow-alt-circle-down')
          .classList.add('invisible');
        document
          .querySelector('.project__seeMore-down')
          .classList.add('invisible');
      }
    } else {
      if (this.state.showThumbnails) {
        if (document.querySelector('.projects__list').scrollTop >= 2000) {
          document
            .querySelector('.fa-arrow-alt-circle-down')
            .classList.add('invisible');
          document
            .querySelector('.project__seeMore-down')
            .classList.add('invisible');
        }
      } else {
        if (
          document.querySelector('.projects__list').scrollTop >=
          1488 * adjustments
        ) {
          document
            .querySelector('.fa-arrow-alt-circle-down')
            .classList.add('invisible');
          document
            .querySelector('.project__seeMore-down')
            .classList.add('invisible');
        }
      }
    }
  };

  previousProjects = () => {
    let adjustments = 1;
    if (this.state.showThumbnails) {
      adjustments = 2;
    }

    if (window.matchMedia('(min-width: 768px)').matches) {
      document.querySelector('.projects__list').scrollTop -= 252;
    } else {
      if (this.state.showThumbnails) {
        document.querySelector('.projects__list').scrollTop -=
          195 * adjustments;
      } else {
        document.querySelector('.projects__list').scrollTop -=
          248 * adjustments;
      }
    }

    document
      .querySelector('.fa-arrow-alt-circle-down')
      .classList.remove('invisible');
    document
      .querySelector('.project__seeMore-down')
      .classList.remove('invisible');

    if (document.querySelector('.projects__list').scrollTop === 0) {
      document
        .querySelector('.fa-arrow-alt-circle-up')
        .classList.add('invisible');
      document.querySelector('.project__seeMore-up').classList.add('invisible');
    }
  };

  render() {
    const text = this.props.text;
    const language = this.props.language;

    return (
      <React.Fragment>
        <div className="developer__projects--thumbnails-checkbox">
          <input
            id="developer__projects-checkbox"
            type="checkbox"
            className="developer__projects-checkbox"
            onClick={this.handleThumbnails}
          />
          <label htmlFor="developer__projects-checkbox">
            {text[language].showThumbnails}
          </label>
        </div>
        <ul className="projects__list">
          {projectsDB.map((project, index) => {
            return (
              <React.Fragment key={index}>
                <li className="project__item">
                  <a
                    href={project.url}
                    className="project__url-a"
                    target="_Blank"
                  >
                    <h3 className="project__title">
                      {project.title}{' '}
                      <i className="far fa-eye project__title-icon"></i>
                    </h3>
                  </a>
                  <p className="project__description">
                    {project.description[language]}
                  </p>

                  <a
                    href={project.repo}
                    className="project__url-a"
                    target="_Blank"
                  >
                    <p className="project__repo">
                      <i className="fab fa-github-alt project__repo--icon"></i>
                      {project.repo}
                    </p>
                  </a>
                </li>

                {this.state.showThumbnails ? (
                  <a
                    href={project.url}
                    className="project__url-a"
                    target="_Blank"
                  >
                    <img
                      src={project.thumbnail}
                      alt={project.title}
                      className="project__thumbnail"
                    />
                  </a>
                ) : null}
              </React.Fragment>
            );
          })}
        </ul>
        {this.props.seeAll ? null : (
          <div className="project__seeMore">
            <span className="project__seeMore-text project__seeMore-up invisible">
              {text[language].goUp}
            </span>
            <i
              className="far fa-arrow-alt-circle-up icon--more invisible"
              onClick={this.previousProjects}
            ></i>
            <i
              className="far fa-arrow-alt-circle-down icon--more"
              onClick={this.nextProjects}
            ></i>
            <span className="project__seeMore-text project__seeMore-down">
              {text[language].showMore}
            </span>
          </div>
        )}
      </React.Fragment>
    );
  }
}

export default ProjectsList;
