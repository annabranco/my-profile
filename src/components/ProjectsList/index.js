import React from 'react';
import { useSelector } from 'react-redux';
import { arrayOf, bool, func, number, oneOf } from 'prop-types';
import ProjectDetails from '../ProjectDetails';
import { developerTextSelector } from '../../redux/selectors';
import {
  ADVANCE_ACTION,
  BACK_ACTION,
  SHOW_THUMBNAILS_ACTION,
  SHOW_THUMBNAILS_ON_MOBILE_ACTION
} from '../../constants';
import {
  Checkbox,
  CheckboxWrapper,
  Icon,
  Paginator,
  ProjectsGrid,
  ProjectsSection,
  Text
} from './styles';
import { projectsPropType } from '../../types';

const ProjectsList = ({
  actualPage,
  displayThumbnails,
  onClickChangePage,
  projects,
  thumbnailsStyle,
  toggleProjectsThumbNails,
  totalPages
}) => {
  const texts = useSelector(developerTextSelector);

  return (
    <ProjectsSection>
      <CheckboxWrapper>
        <Checkbox
          id="developer__projects-checkbox"
          onClick={event =>
            toggleProjectsThumbNails(event.currentTarget.checked)
          }
          type="checkbox"
        />
        <label htmlFor="developer__projects-checkbox">
          {texts.showThumbnails}
        </label>
      </CheckboxWrapper>
      <ProjectsGrid thumbnailsStyle={thumbnailsStyle}>
        {projects &&
          projects[actualPage - 1].map(project => (
            <ProjectDetails
              displayThumbnails={displayThumbnails}
              key={project.title}
              project={project}
            />
          ))}
      </ProjectsGrid>
      <Paginator>
        <Text hidden={actualPage === 1}>{texts.goUp}</Text>
        <Icon
          aria-label={texts.goUp}
          className="far fa-arrow-alt-circle-up"
          hidden={actualPage === 1}
          onClick={() => onClickChangePage(BACK_ACTION)}
          role="button"
          tabIndex={0}
        />
        <Icon
          aria-label={texts.showMore}
          className="far fa-arrow-alt-circle-down"
          hidden={actualPage === totalPages}
          onClick={() => onClickChangePage(ADVANCE_ACTION)}
          role="button"
          tabIndex={0}
        />
        <Text hidden={actualPage === totalPages}>{texts.showMore}</Text>
      </Paginator>
    </ProjectsSection>
  );
};

ProjectsList.propTypes = {
  actualPage: number.isRequired,
  displayThumbnails: bool.isRequired,
  onClickChangePage: func.isRequired,
  projects: arrayOf(arrayOf(projectsPropType)),
  thumbnailsStyle: oneOf([
    SHOW_THUMBNAILS_ON_MOBILE_ACTION,
    SHOW_THUMBNAILS_ACTION
  ]),
  toggleProjectsThumbNails: func.isRequired,
  totalPages: number.isRequired
};

ProjectsList.defaultProps = {
  thumbnailsStyle: ''
};

ProjectsList.defaultProps = {
  projects: [
    [
      {
        description: {
          en: '',
          es: '',
          pt: ''
        },
        order: 0,
        repo: '',
        thumbnail: '',
        url: ''
      }
    ]
  ]
};

export default ProjectsList;
