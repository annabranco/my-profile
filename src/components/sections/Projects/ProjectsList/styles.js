import styled, { css } from 'styled-components';
import {
  SHOW_THUMBNAILS_ACTION,
  SHOW_THUMBNAILS_ON_MOBILE_ACTION
} from '../../../../constants';
import { Hidden } from '../../../../styles/global';

export const Checkbox = styled.input`
  margin: 0 10px;
`;
Checkbox.displayName = 'Checkbox';

export const CheckboxWrapper = styled.div`
  width: 100%;
  margin: 30px auto;
  text-align: center;
`;
CheckboxWrapper.displayName = 'Checkbox Area';

export const Icon = styled.i`
  margin: 0 5px;
  font-size: 2rem;
  cursor: pointer;
  outline: none;
  -webkit-text-stroke: 1px black;

  &:hover {
    color: orange;
  }
  ${({ notVisible }) =>
    notVisible &&
    css`
      ${Hidden}
    `}
`;
Icon.displayName = 'Icon';

export const Paginator = styled.div`
  opacity: 0.5;
  margin-bottom: 40px;
  text-align: center;
`;
Paginator.displayName = 'Paginator';

export const ProjectsGrid = styled.ul`
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 225px;
  grid-gap: 30px 40px;
  height: 210px;

  @media all and (min-width: 768px) {
    grid-template-columns: 1fr 1fr;
    grid-template-rows: 360px;
  }

  ${({ thumbnailsStyle }) => {
    if (thumbnailsStyle === SHOW_THUMBNAILS_ON_MOBILE_ACTION) {
      return css`
        grid-template-rows: 360px;
        height: 380px;
      `;
    }
    if (thumbnailsStyle === SHOW_THUMBNAILS_ACTION) {
      return css`
        height: 390px;
      `;
    }
    return null;
  }}
`;
ProjectsGrid.displayName = 'Projects Grid';

export const ProjectsSection = styled.section`
  padding-top: 20px;
  width: 60%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;
ProjectsSection.displayName = 'PROJECTS Section';

export const Text = styled.span`
  vertical-align: super;
  font-size: 0.8rem;
  text-align: center;

  ${({ notVisible }) =>
    notVisible &&
    css`
      ${Hidden}
    `}
`;
Text.displayName = 'Text';
