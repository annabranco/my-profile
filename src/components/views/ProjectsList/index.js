import React, { Fragment } from 'react';
import projectsDB from '../../../db/projectsDB';

export const ProjectsList = ({
  texts,
  language,
  toggleProjectsThumbNails,
  onClickNextProjects,
  onClickPreviousProjects,
  seeAllProjects,
  displayThumbnails
}) => (
  <Fragment>
    <div className="developer__projects--thumbnails-checkbox">
      <input
        id="developer__projects-checkbox"
        type="checkbox"
        className="developer__projects-checkbox"
        onClick={event => toggleProjectsThumbNails(event.currentTarget.checked)}
      />
      <label htmlFor="developer__projects-checkbox">
        {texts.showThumbnails}
      </label>
    </div>
    <ul className="projects__list">
      {projectsDB.map((project, index) => (
        <Fragment key={index}>
          <li className="project__item">
            <a href={project.url} className="project__url-a" target="_Blank">
              <h3 className="project__title">
                {project.title}{' '}
                <i className="far fa-eye project__title-icon"></i>
              </h3>
            </a>
            <p className="project__description">
              {project.description[language]}
            </p>

            <a href={project.repo} className="project__url-a" target="_Blank">
              <p className="project__repo">
                <i className="fab fa-github-alt project__repo--icon"></i>
                {project.repo}
              </p>
            </a>
          </li>

          {displayThumbnails && (
            <a href={project.url} className="project__url-a" target="_Blank">
              <img
                src={project.thumbnail}
                alt={project.title}
                className="project__thumbnail"
              />
            </a>
          )}
        </Fragment>
      ))}
    </ul>
    {!seeAllProjects && (
      <div className="project__seeMore">
        <span className="project__seeMore-text project__seeMore-up invisible">
          {texts.goUp}
        </span>
        <i
          className="far fa-arrow-alt-circle-up icon--more invisible"
          onClick={onClickPreviousProjects}
        ></i>
        <i
          className="far fa-arrow-alt-circle-down icon--more"
          onClick={onClickNextProjects}
        ></i>
        <span className="project__seeMore-text project__seeMore-down">
          {texts.showMore}
        </span>
      </div>
    )}
  </Fragment>
);
