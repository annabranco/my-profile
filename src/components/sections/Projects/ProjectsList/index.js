import React from 'react';
import { useSelector } from 'react-redux';
import { arrayOf, bool, func, number, oneOf } from 'prop-types';
import { developerTextSelector } from '../../../../redux/selectors';
import ProjectDetails from '../ProjectDetails';
import {
  ADVANCE_ACTION,
  BACK_ACTION,
  SHOW_THUMBNAILS_ACTION,
  SHOW_THUMBNAILS_ON_MOBILE_ACTION
} from '../../../../constants';
import { projectsPropType } from '../../../../types';
import { Icon, Paginator, ProjectsGrid, ProjectsSection, Text } from './styles';
import { isDesktop } from '../../../../utils/device';

const ProjectsList = ({
  actualPage,
  onClickChangePage,
  projects,
  thumbnailsStyle,
  totalPages,
  visible
}) => {
  const texts = useSelector(developerTextSelector);

  return (
    <ProjectsSection>
      {(visible || !isDesktop) && (
        <>
          <ProjectsGrid thumbnailsStyle={thumbnailsStyle}>
            {projects &&
              projects[actualPage - 1].map(project => (
                <ProjectDetails
                  actualPage={actualPage}
                  key={project.title}
                  project={project}
                />
              ))}
          </ProjectsGrid>

          <Paginator>
            <Text notVisible={actualPage === 1}>{texts.goUp}</Text>
            <Icon
              aria-label={texts.goUp}
              className="far fa-arrow-alt-circle-up"
              notVisible={actualPage === 1}
              onClick={() => onClickChangePage(BACK_ACTION)}
              role="button"
              tabIndex={0}
            />
            <Icon
              aria-label={texts.showMore}
              className="far fa-arrow-alt-circle-down"
              next
              notVisible={actualPage === totalPages}
              onClick={() => onClickChangePage(ADVANCE_ACTION)}
              role="button"
              tabIndex={0}
            />
            <Text next notVisible={actualPage === totalPages}>
              {texts.showMore}
            </Text>
          </Paginator>
        </>
      )}
    </ProjectsSection>
  );
};

ProjectsList.propTypes = {
  actualPage: number.isRequired,
  onClickChangePage: func.isRequired,
  projects: arrayOf(arrayOf(projectsPropType)),
  thumbnailsStyle: oneOf([
    SHOW_THUMBNAILS_ON_MOBILE_ACTION,
    SHOW_THUMBNAILS_ACTION
  ]),
  totalPages: number.isRequired,
  visible: bool.isRequired
};

ProjectsList.defaultProps = {
  projects: [
    [
      {
        description: {
          en: '',
          es: '',
          pt: ''
        },
        order: 0,
        repo: '',
        thumbnail: '',
        url: ''
      }
    ]
  ],
  thumbnailsStyle: ''
};

export default ProjectsList;
