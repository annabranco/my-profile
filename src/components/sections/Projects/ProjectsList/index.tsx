import React, { ReactElement } from 'react';
import { useSelector } from 'react-redux';
import { arrayOf, bool, func, number } from 'prop-types';
import { IProjects } from '../../../../types/interfaces';
import { developerTextSelector } from '../../../../redux/selectors';
import { isDesktop } from '../../../../utils/device';
import ProjectDetails from '../ProjectDetails';
import { ADVANCE_ACTION, BACK_ACTION } from '../../../../constants';
import { projectsPropType } from '../../../../types/propTypes';
import { Icon, Paginator, ProjectsGrid, ProjectsSection, Text } from './styles';

interface Props {
  actualPage: number;
  onClickChangePage: (action: string) => void;
  projects: IProjects[][];
  totalPages: number;
  visible: boolean;
}

const ProjectsList = ({
  actualPage,
  onClickChangePage,
  projects,
  totalPages,
  visible
}: Props): ReactElement => {
  const texts = useSelector(developerTextSelector);

  return (
    <ProjectsSection>
      {(visible || !isDesktop) && (
        <>
          <ProjectsGrid>
            {projects &&
              projects[actualPage - 1].map(project => (
                <ProjectDetails
                  actualPage={actualPage}
                  key={project.title}
                  project={project}
                />
              ))}
          </ProjectsGrid>

          <Paginator>
            <Text notVisible={actualPage === 1}>{texts.goUp}</Text>
            <Icon
              aria-label={texts.goUp}
              className="far fa-arrow-alt-circle-up"
              notVisible={actualPage === 1}
              onClick={() => onClickChangePage(BACK_ACTION)}
              role="button"
              tabIndex={0}
            />
            <Icon
              aria-label={texts.showMore}
              className="far fa-arrow-alt-circle-down"
              next
              notVisible={actualPage === totalPages}
              onClick={() => onClickChangePage(ADVANCE_ACTION)}
              role="button"
              tabIndex={0}
            />
            <Text next notVisible={actualPage === totalPages}>
              {texts.showMore}
            </Text>
          </Paginator>
        </>
      )}
    </ProjectsSection>
  );
};

ProjectsList.propTypes = {
  actualPage: number.isRequired,
  onClickChangePage: func.isRequired,
  projects: arrayOf(arrayOf(projectsPropType)),
  totalPages: number.isRequired,
  visible: bool.isRequired
};

ProjectsList.defaultProps = {
  projects: [
    [
      {
        description: {
          en: '',
          es: '',
          pt: ''
        },
        order: 0,
        repo: '',
        thumbnail: '',
        url: ''
      }
    ]
  ]
};

export default ProjectsList;
