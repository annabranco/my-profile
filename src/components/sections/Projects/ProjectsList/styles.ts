import styled, { css } from 'styled-components';
import { Hidden } from '../../../../styles/global';

interface IconProps extends React.ComponentPropsWithoutRef<'i'> {
  notVisible?: boolean;
  next?: boolean;
}

export const Icon = styled.i<IconProps>`
  position: absolute;
  bottom: 40vh;
  font-size: 2.5rem;
  cursor: pointer;
  outline: none;
  -webkit-text-stroke: 1px black;

  &:hover {
    color: orange;
  }
  ${({ notVisible }) =>
    notVisible &&
    css`
      ${Hidden}
    `}

  ${({ next }) => {
    if (next) {
      return css`
        right: 3px;
      `;
    }
    return css`
      left: 3px;
    `;
  }}

  @media all and (min-width: 768px) {
    position: unset;
    margin: 0 5px;
    transform: none;
    font-size: 2rem;
  }
`;
Icon.displayName = 'Icon';

export const Paginator = styled.div`
  height: 100vh;
  width: 100vw;
  opacity: 0.5;
  text-align: center;

  @media all and (min-width: 768px) {
    height: auto;
    width: auto;
    margin: 30px auto;
  }
`;
Paginator.displayName = 'Paginator';

export const ProjectsGrid = styled.ul`
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 50% 50%;
  grid-gap: 15px;
  height: 70%;

  @media all and (min-width: 768px) {
    grid-gap: 30px 40px;
    grid-template-columns: 1fr 1fr;
    grid-template-rows: auto;
  }
`;
ProjectsGrid.displayName = 'Projects Grid';

export const ProjectsSection = styled.section`
  padding-top: 8vh;
  width: 80%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  transition: height 3s linear;

  @media all and (min-width: 375px) {
    padding-top: 10vh;
  }

  @media all and (min-width: 768px) {
    padding-top: 20px;
    width: 60%;
    min-height: 35vw;
  }
`;
ProjectsSection.displayName = 'PROJECTS Section';

interface TextProps extends React.ComponentPropsWithoutRef<'span'> {
  notVisible?: boolean;
  next?: boolean;
}

export const Text = styled.span<TextProps>`
  position: absolute;
  vertical-align: super;
  font-size: 1rem;
  text-align: center;
  writing-mode: vertical-lr;
  transform: rotate(180deg);

  ${({ notVisible }) =>
    notVisible &&
    css`
      ${Hidden}
    `}

  ${({ next }) => {
    if (next) {
      return css`
        bottom: 40vh;
        right: 15px;
        transform: rotate(180deg) translate(0, 50px);
      `;
    }
    return css`
      bottom: 28vh;
      left: 15px;
      transform: translate(0, 50px);
    `;
  }}
  @media all and (min-width: 768px) {
    position: unset;
    font-size: 0.8rem;
    writing-mode: lr;
    transform: none;
  }
`;
Text.displayName = 'Text';
