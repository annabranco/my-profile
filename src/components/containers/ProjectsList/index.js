import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import chunk from 'lodash/chunk';
import sortBy from 'lodash/sortBy';
import { ProjectDetails } from '../../views';
import {
  SHOW_THUMBNAILS_ACTION,
  HIDE_ACTION,
  ADVANCE_ACTION,
  BACK_ACTION,
  SHOW_THUMBNAILS_ON_MOBILE_ACTION
} from '../../../constants';
import { developerTextPropType, projectsPropType } from '../../../types';
import { isDesktop } from '../../../utils/device';

class ProjectsList extends Component {
  static propTypes = {
    texts: developerTextPropType.isRequired,
    projects: PropTypes.arrayOf(projectsPropType).isRequired,
    language: PropTypes.string.isRequired
    // handleAdjustExpandedProjectsView: PropTypes.func.isRequired
  };

  state = {
    displayThumbnails: false,
    actualPage: 1,
    totalPages: 0,
    projectsList: [[]]
  };

  componentDidMount() {
    this.organizeProjectsList(isDesktop ? 2 : 1);
  }

  componentDidUpdate(prevProps, prevState) {
    if (
      prevState.knowMore !== this.state.knowMore ||
      prevState.displayThumbnails !== this.state.displayThumbnails
    ) {
      let adjust = 0;
      if (this.state.displayThumbnails) {
        adjust += 150;
      }
      if (this.state.knowMore) {
        adjust += 200;
      }
      // this.props.handleAdjustExpandedProjectsView(adjust);
    }
  }

  organizeProjectsList = numberOfProjectsPerPage => {
    const organizedProjects = chunk(
      sortBy(this.props.projects),
      numberOfProjectsPerPage
    );
    this.setState({
      projectsList: organizedProjects,
      totalPages: organizedProjects.length
    });
  };

  toggleProjectsThumbNails = isVisible =>
    this.setState({
      displayThumbnails: isVisible
    });

  onClickChangePage = action => {
    const { actualPage } = this.state;
    let nextPage;

    if (action === ADVANCE_ACTION) {
      nextPage = actualPage + 1;
    } else if (action === BACK_ACTION) {
      nextPage = actualPage - 1;
    }

    if (nextPage) {
      this.setState({
        actualPage: nextPage
      });
    }
  };

  setProjectsListClasses = () => {
    const { displayThumbnails } = this.state;

    if (displayThumbnails && isDesktop) {
      return ` ${SHOW_THUMBNAILS_ACTION}`;
    }
    if (displayThumbnails) {
      return ` ${SHOW_THUMBNAILS_ON_MOBILE_ACTION}`;
    }
    return '';
  };

  render() {
    const {
      displayThumbnails,
      projectsList,
      actualPage,
      totalPages
    } = this.state;
    const { texts, language } = this.props;

    return (
      <section>
        <div className="developer__projects--thumbnails-checkbox">
          <input
            id="developer__projects-checkbox"
            type="checkbox"
            className="developer__projects-checkbox"
            onClick={event =>
              this.toggleProjectsThumbNails(event.currentTarget.checked)
            }
          />
          <label htmlFor="developer__projects-checkbox">
            {texts.showThumbnails}
          </label>
        </div>
        <ul className={`projects__list${this.setProjectsListClasses()}`}>
          {projectsList[actualPage - 1].map(project => (
            <ProjectDetails
              project={project}
              language={language}
              displayThumbnails={displayThumbnails}
              key={project.title}
            />
          ))}
        </ul>
        <div className="project__seeMore">
          <span
            className={`project__seeMore-text project__seeMore-up ${actualPage ===
              1 && HIDE_ACTION}`}
          >
            {texts.goUp}
          </span>
          <i
            className={`far fa-arrow-alt-circle-up icon--more ${actualPage ===
              1 && HIDE_ACTION}`}
            onClick={() => this.onClickChangePage(BACK_ACTION)}
            role="button"
            aria-label={texts.goUp}
            tabIndex={0}
          />
          <i
            className={`far fa-arrow-alt-circle-down icon--more ${actualPage ===
              totalPages && HIDE_ACTION}`}
            onClick={() => this.onClickChangePage(ADVANCE_ACTION)}
            role="button"
            aria-label={texts.showMore}
            tabIndex={0}
          />
          <span
            className={`project__seeMore-text project__seeMore-down ${actualPage ===
              totalPages && HIDE_ACTION}`}
          >
            {texts.showMore}
          </span>
        </div>
      </section>
    );
  }
}

export default ProjectsList;
