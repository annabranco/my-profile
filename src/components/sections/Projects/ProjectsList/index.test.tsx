import { shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';
import chunk from 'lodash/chunk';
import sortBy from 'lodash/sortBy';
import ProjectsList from '.';
import { getMockState, setupMockProvider } from '../../../../../testing/mocks';

// TODO [07-May -21]: Define real test suits (Anna Branco)

describe('< ProjectsList >', () => {
  const mockProjects = getMockState({
    language: 'en'
  }).projects;
  const pagedMockProjects = chunk(sortBy(mockProjects), 4);
  let MockProvider;
  let wrapper: ShallowWrapper;

  beforeAll(() => {
    MockProvider = setupMockProvider({ language: 'en' }).MockProvider;
    wrapper = shallow(
      <MockProvider>
        <ProjectsList
          actualPage={1}
          onClickChangePage={() => jest.fn()}
          projects={pagedMockProjects}
          totalPages={4}
          visible
        />
      </MockProvider>
    );
  });

  it('should mount', () => {
    expect(wrapper).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
