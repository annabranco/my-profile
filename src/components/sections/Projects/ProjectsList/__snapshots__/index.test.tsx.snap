// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< ProjectsList > matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ProjectsList
    actualPage={1}
    onClickChangePage={[Function]}
    projects={
      Array [
        Array [
          Object {
            "description": Object {
              "en": "An  unofficial app to enhance players' experience while playing Zombicide tabletop game.",
              "es": "Una app no oficial para mejorar la experiencia del juego de mesa Zombicide.",
              "pt": "Una app nãp oficial para melhorar a experiência do jogo de tabuleiro Zombicide.",
            },
            "order": 1,
            "repo": "https://github.com/annabranco/zombicide-party",
            "thumbnail": "zombicide-party",
            "title": "Zombicide Party",
            "url": "https://annabranco.github.io/zombicide-party/",
          },
          Object {
            "description": Object {
              "en": "A page for No Name Kitchen, a NGO working with refugees arriving in Europe. Pre-release.",
              "es": "Una página para No Name Kitchen, una ONG que trabaja con refugiados. Pre-release.",
              "pt": "Uma página para No Name Kitchen, uma ONG que trabalha com refugiados. Pre-release.",
            },
            "order": 2,
            "repo": "https://github.com/annabranco/nnk",
            "thumbnail": "no-name-kitchen",
            "title": "No Name Kitchen",
            "url": "http://nnk.vercel.app/",
          },
          Object {
            "description": Object {
              "en": "A fun and beautiful memory card game themed on Disney’s Toy Story characters, for all ages.",
              "es": "Un divertido y bonito juego de parejas, con tema de Toy Story (Disney), para todas las edades. ",
              "pt": "Um divertido e bonito jogo da memória, com temática de Toy Story (Disney), para todas as idades.",
            },
            "order": 3,
            "repo": "https://github.com/annabranco/toy-story-games",
            "thumbnail": "toy-story",
            "title": "Toy Story Games",
            "url": "https://annabranco.github.io/toy-story-games/",
          },
          Object {
            "description": Object {
              "en": "The prototype of an own Role Playing Game built in vanilla JavaScript. Work in progress.",
              "es": "El prototipo de un juego de rol mío desarrollado en JavaScript vanilla. En desarrollo.",
              "pt": "O protótipo de um jogo de RPG meu programado en JavaScript vanilla. Em desenvolvimento.",
            },
            "order": 4,
            "repo": "https://github.com/annabranco/an-adventure",
            "thumbnail": "an-adventure",
            "title": "An Adventure",
            "url": "https://annabranco.github.io/an-adventure/",
          },
        ],
        Array [
          Object {
            "description": Object {
              "en": "A demo pre-release of a game of Shougi, the Japanese Chess. Unfinished, but playable.",
              "es": "Una demo pre-release de un juego de Shougi, el ajedrez japonés. No está terminado.",
              "pt": "Uma demo pre-release de um jogo de Shougi, o xadrez japonês. Não está terminado.",
            },
            "order": 5,
            "repo": "https://github.com/annabranco/shougi",
            "thumbnail": "shougi",
            "title": "Shougi",
            "url": "https://annabranco.github.io/shougi/",
          },
          Object {
            "description": Object {
              "en": "A beautiful visual database presentating Harry Potter's characters.",
              "es": "Un bonito proyecto visual presentando los personajes de Harry Potter.",
              "pt": "Un bonito projeto visual apresetnando os personagens de Harry Potter.",
            },
            "order": 6,
            "repo": "https://github.com/annabranco/harry-potter-database",
            "thumbnail": "harry-potter-database",
            "title": "Harry Potter Database",
            "url": "https://annabranco.github.io/harry-potter-database",
          },
          Object {
            "description": Object {
              "en": "A beautiful TV series finder, where users can search on a huge database.",
              "es": "Un buscador de series, donde puedes buscar en una enorme base de datos.",
              "pt": "Um buscador de séries, onde você pode buscar en um enorme banco de dados.",
            },
            "order": 7,
            "repo": "https://github.com/annabranco/tv-series-finder",
            "thumbnail": "tv-series-finder",
            "title": "TV Series Finder",
            "url": "https://annabranco.github.io/tv-series-finder/",
          },
          Object {
            "description": Object {
              "en": "This is the page you are visiting; a digital presentation of CV and professional profile.",
              "es": "Esta es la página donde estás; una presentación digital de mi CV y perfil profesional.",
              "pt": "Esta é a página onde você está; uma apresentação digital do meu CV e perfil profissional.",
            },
            "order": 8,
            "repo": "https://github.com/annabranco/my-profile",
            "thumbnail": "my-profile",
            "title": "My profile",
            "url": "https://annabranco.github.io/my-profile/",
          },
        ],
        Array [
          Object {
            "description": Object {
              "en": "An animation game where the player interacts with the weather. In progress.",
              "es": "Un juego de animación donde el jugador interactúa con el tiempo. En desarrollo.",
              "pt": "Um jogo de animação onde o jogador interage con o tempo. Em desenvolvimento.",
            },
            "order": 9,
            "repo": "https://github.com/annabranco/days-go-by",
            "thumbnail": "days-go-by",
            "title": "Days Go By",
            "url": "https://annabranco.github.io/days-go-by/",
          },
        ],
      ]
    }
    totalPages={4}
    visible={true}
  />
</Provider>
`;
